pkg/cmd/pr/shared/editable.go

```diff
@@ -0,0 +1,310 @@
+	"github.com/cli/cli/internal/ghrepo"
+	"github.com/cli/cli/pkg/surveyext"
+)
+
+type EditableOptions struct {
+	Title        string
+	TitleDefault string
+	TitleEdited  bool
+
+	Body        string
+	BodyDefault string
+	BodyEdited  bool
+
+	Reviewers        []string
+	ReviewersDefault api.ReviewRequests
+	ReviewersOptions []string
+	ReviewersEdited  bool
+	ReviewersAllowed bool
+
+	Assignees        []string
+	AssigneesDefault api.Assignees
+	AssigneesOptions []string
+	AssigneesEdited  bool
+
+	Labels        []string
+	LabelsDefault api.Labels
+	LabelsOptions []string
+	LabelsEdited  bool
+
+	Projects        []string
+	ProjectsDefault api.ProjectCards
+	ProjectsOptions []string
+	ProjectsEdited  bool
+
+	Milestone        string
+	MilestoneDefault api.Milestone
+	MilestoneOptions []string
+	MilestoneEdited  bool
+
+	Metadata api.RepoMetadataResult
+}
+
+func (e EditableOptions) Dirty() bool {
+	return e.TitleEdited ||
+		e.BodyEdited ||
+		e.ReviewersEdited ||
+		e.AssigneesEdited ||
+		e.LabelsEdited ||
+		e.ProjectsEdited ||
+		e.MilestoneEdited
+}
+
+func EditableSurvey(editorCommand string, options *EditableOptions) error {
+	if options.TitleEdited {
+		title, err := titleSurvey(options.TitleDefault)
+		if err != nil {
+			return err
+		}
+		options.Title = title
+	}
+	if options.BodyEdited {
+		body, err := bodySurvey(options.BodyDefault, editorCommand)
+		if err != nil {
+			return err
+		}
+		options.Body = body
+	}
+	if options.AssigneesEdited {
+		assignees, err := assigneesSurvey(options.AssigneesDefault, options.AssigneesOptions)
+		if err != nil {
+			return err
+		}
+		options.Assignees = assignees
+	}
+	if options.LabelsEdited {
+		labels, err := labelsSurvey(options.LabelsDefault, options.LabelsOptions)
+		if err != nil {
+			return err
+		}
+		options.Labels = labels
+	}
+	if options.ProjectsEdited {
+		projects, err := projectsSurvey(options.ProjectsDefault, options.ProjectsOptions)
+		if err != nil {
+			return err
+		}
+		options.Projects = projects
+	}
+	if options.MilestoneEdited {
+		milestone, err := milestoneSurvey(options.MilestoneDefault, options.MilestoneOptions)
+		if err != nil {
+			return err
+		}
+		options.Milestone = milestone
+	}
+	confirm, err := confirmSurvey()
+	if err != nil {
+		return err
+	}
+	if !confirm {
+		return fmt.Errorf("Discarding...")
+	}
+
+	return nil
+}
+
+func FieldsToEditSurvey(options *EditableOptions) error {
+	contains := func(s []string, str string) bool {
+		for _, v := range s {
+			if v == str {
+				return true
+			}
+		}
+		return false
+	}
+
+	results := []string{}
+	opts := []string{"Title", "Body"}
+	if options.ReviewersAllowed {
+		opts = append(opts, "Reviewers")
+	}
+	opts = append(opts, "Assignees", "Labels", "Projects", "Milestone")
+	q := &survey.MultiSelect{
+		Message: "What would you like to edit?",
+		Options: opts,
+	}
+	err := survey.AskOne(q, &results)
+	if err != nil {
+		return err
+	}
+
+	if contains(results, "Title") {
+		options.TitleEdited = true
```

Comment:
Tiny naming oddity: it's a bit weird that simply declaring your intent that you want to edit the title would set the `opts.TitleEdited` field to true, and that the same field is used when `--title` was passed on the command-line. But, I guess it works!